app/main/posts/common/post-actions.directive.js

```diff
@@ -75,7 +77,7 @@ function PostActionsDirective(
             } else {
                 $scope.selectedPost = {post: $scope.post};
             }
-            if ($location.path().indexOf('data') === -1) {
+            if ($routeParams.view !== 'data') {
```

Comment:
Using indexOf covers saved-searches and collections too. If we only check for view and select a post in saved-search or collection-mode, the post opens in post-detail-view

Reply:
yes, that makes sense. I'll look into it. Thanks Anna